### ISSUE TITLE
Instances (2 Issues).

### ISSUE THREAD

**Issue by UltimateMors** [ronancpl/MapleSolaxiaV2#117]
There is a very odd but not very serious bugs involving instances. 

**Number 1**: One of my players discovered, if you crashed during a (I assume any PQ), AgentPQ, come back, you will for some reason no longer gain EXP from killing anything until you leave the party. 

To emulate a crash, I used !debugpacket by itself, which causes a crash, while in AgentPQ. Going back in game, I spawned a single snail, killed it, gained no EXP. Went to Henesys Hunting Grounds, made sure being in hide didn't cause any issues, and it was the same issue. I had gained no exp from mobs. **Oddly, if you change channel, or, just simply leave the party, everything will be fine, you will gain EXP normally.**

**Number 2**: This also leads into an odd issue if you leave the AgentPQ, and then try to re-enter, the entrance NPC will explain that someone is already in the Party Quest. **If you change channel, or,  once again, simply leave the party, everything will be fine**. Going to cash shop does not fix the issue, and changing maps will not fix the issue as well. 

Thanks for looking into it if you can. Instance issues were the main reason I stopped working on v146, because I'm a bad coder and this source is teaching me a lot. Haha.

**Comment by ronancpl** [comment]
This Number 2 seems like you tried to immediately reenter the event after cancelling/completing the event. There is a thing on the source: I made a cooldown-of-sorts system for the events. After the event dispose function is called, there is a 15sec timeout before accepting new instances. Hence, in the mean time, it may seems somebody is PQ'ing when actually they all are waiting for the reopening. Other than that, I'm afraid I couldn't reproduce this issue.

Yea, I've worked my way to make the event instances work neatly. That was not an easy task as a lot of "PQ thingies" seemed to be scattered all over the places, each script implementing the same thing on their own. I've managed to abstract a lot of these funcionalities (event timeout, event exclusive items, stage EXP, you name it) and had them covered inside the EIM core, thus making the scripts cleaner and more reliable for expected behaviours.
